back/src/main/java/com/ubikloadpack/jmeter/ulp/observability/registry/MicrometerRegistry.java

```diff
@@ -213,11 +213,15 @@ public SampleLog makeLog(String name, Date timestamp) {
 		
 		Double averageTotalResponseTime = summaryRegistry.counter("accumulate.response", "sample", name).count() /
 				summaryRegistry.counter("count.total", "sample", name).count();
-		double timeSinceFirstSampleCallInSeconds = (startAndEndDatesOfSamplers.get(name).getRight() - startAndEndDatesOfSamplers.get(name).getLeft()) / 1000d; 
+		
+		double timeSinceFirstSampleCallInSeconds = 0;
+		if (startAndEndDatesOfSamplers.containsKey(name)) {
```

Comment:
Instead of doing it like this:
Get the value by key : startAndEndDatesOfSamplers.get(name)
Then check for non nullity

Currently there are 3 access in Hashmap instead of 1